mergesort.py

```diff
@@ -0,0 +1,98 @@
+#! /usr/bin/env python
+import time
+from bst import Bst
+from list import makeworstlist
+
+
+
+def timed_func(func):
+    """Decorator for timing our insertion sort methods."""
+    def timed(*args, **kwargs):
+        start = time.time()
+        result = func(*args, **kwargs)
+        elapsed = time.time() - start
+        print "time expired: %s" % elapsed
+        return result
+    return timed
+
+
+def merge_sort(l):
+    """Sorts a list using a simple merge sort implemetnation"""
+    if len(l) <= 1:
+        return l
+    else:
+        left = l[:len(l)/2]
+        right = l[len(l)/2:]
+
+        left = merge_sort(left)
+        right = merge_sort(right)
+
+    return merge(left, right)
+
+
+def merge(left, right):
+    """Combines two lists lowest to greates
+    is a stable merge method"""
+    result = []
+    while left and right:
+        if left[0] <= right[0]:
+            result.append(left.pop(0))
+        else:
+            result.append(right.pop(0))
+    while left:
+        result.append(left.pop(0))
```

Comment:
Here there's really no reason at all to consume the `left` or `right` lists one item at a time.  You are just slapping whatever is in those lists directly onto the end of the existing result.